app/src/main/java/com/example/neighborly/FeedFragment.java

```diff
@@ -1,36 +1,63 @@
 package com.example.neighborly;
 
+import android.app.Dialog;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
 import android.os.Bundle;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
+import android.widget.EditText;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
-import androidx.viewpager.widget.ViewPager;
 
 import java.util.ArrayList;
 
 public class FeedFragment extends Fragment {
-
-    Integer[] colors = null;
-    ArrayList<RequestModel> privateChats;
-    ViewPager privateChatCarouselViewPager;
-    PrivateChatAdapter privateChatCarouselAdapter;
-    int privateChatCarouselPosition;
+    View feedView;
+    Dialog popupRequestDialog;
 
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
-        View view = inflater.inflate(R.layout.fragment_feed, container, false);
-        Button searchButton = view.findViewById(R.id.searchButton);
-        return view;
+        super.onCreate(savedInstanceState);
+        popupRequestDialog = new Dialog(this.getContext());
+        feedView = inflater.inflate(R.layout.fragment_feed, container, false);
+
+        Button searchButton = feedView.findViewById(R.id.searchButton);
+
+        searchButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                EditText searchText = feedView.findViewById(R.id.SearchText);
+                ArrayList<ItemModel> n = searchForItem(searchText.getText().toString());
```

Comment:
Maybe rename this to searchText ?
Plus In the future, we should add a function that "cleans" the search text (removes caps, removes spaces before and after etc). This function should also be called when we send a item to be stored in the DB - in the addItemActivity).